runtime/proto/CorfuProtocol.proto

```diff
@@ -0,0 +1,825 @@
+}
+
+message LogAddressSpaceRequest {
+}
+
+message LogAddressSpaceResponse {
+  int64 logTail = 1;
+  // List of UUIDToStreamAddressPairs
+  List addressMap = 2;
+}
+
+message KnownAddressRequest {
+  int64 startRange = 1;
+  int64 endRange = 2;
+}
+
+message KnownAddressResponse {
+  repeated int64 knownAddresses = 1;
+}
+
+message CommittedTailRequest {
+}
+
+message CommittedTailResponse {
+  int64 committedTail = 1;
+}
+
+message UpdateCommittedTailRequest {
+  int64 committedTail = 1;
+}
+
+message UpdateCommittedTailResponse {
+}
+
+message ResetLogUnitRequest {
+  int64 epoch = 1;
+}
+
+message ResetLogUnitResponse {
+}
+
+/**
+ * Management Definitions
+ **/
+
+message NodeConnectivity {
+  message ConnectivityMapEntry {
+    string key = 1;
+    string value = 2;
+  }
+
+  string endpoint = 1;
+  string connectivityType = 2;
+  List connectivityMap = 3;
+  int64 epoch = 4;
+}
+
+message QueryNodeRequest {
+}
+
+message QueryNodeResponse {
+  SequencerMetrics sequencerMetrics = 1;
+  NodeConnectivity nodeConnectivity = 2;
+}
+
+message ReportFailureRequest {
+  int64 detectorEpoch = 1;
+  repeated string failedNodes = 2;
+}
+
+message ReportFailureResponse {
+  enum Type {
+    ACK = 0;
+    NACK = 1;
+  }
+
+  Type respType = 1;
+}
+
+message HealFailureRequest {
+  int64 detectorEpoch = 1;
+  repeated string healedNodes = 2;
+}
+
+message HealFailureResponse {
+  enum Type {
+    ACK = 0;
+    NACK  = 1;
+  }
+
+  Type respType = 1;
+}
+
+message OrchestratorRequest {
+  enum Type {
+    QUERY_WORKFLOW = 0;
+    ADD_NODE = 1;
+    REMOVE_NODE = 2;
+    HEAL_NODE = 3;
+    FORCE_REMOVE_NODE = 4;
+    RESTORE_REDUNDANCY_MERGE_SEGMENTS = 5;
+  }
+
+  message QueryWorkflow {
+    UUID id = 1;
+  }
+
+  message ExecuteWorkflow {
+    string endpoint = 1;
+    int32  stripeIndex = 2;
+    bool layoutServer = 3;
+    bool sequencerServer = 4;
+    bool logUnitServer = 5;
+  }
+
+  Type reqType = 1;
+
+  oneof payload {
+    QueryWorkflow queryWorkflow = 10;
+    ExecuteWorkflow executeWorkflow = 11;
+  }
+}
+
+message OrchestratorResponse {
+  enum Type {
+    WORKFLOW_STATUS = 0;
+    WORKFLOW_CREATED = 1;
+  }
+
+  message QueryWorkflow {
+    bool isActive = 1;
+  }
+
+  message CreateWorkflow {
+    UUID workflowId = 1;
+  }
+
+  Type respType = 1;
+
+  oneof payload {
+    QueryWorkflow queryWorkflow = 10;
+    CreateWorkflow createWorkflow = 11;
+  }
+}
+
+message BootstrapManagementRequest {
+  Layout layout = 1;
+}
+
+message BootstrapManagementResponse {
+  enum Type {
+    ACK = 0;
+    NACK = 1;
+  }
+
+  Type respType = 1;
+}
+
+message GetManagementLayoutRequest {
+}
+
+message GetManagementLayoutResponse {
+  Layout layout = 1;
+}
+
+/**
+ * Request & Response Definitions
+ **/
+
+message Header {
+  ProtocolVersion version = 1;
+  int64 request_id = 2;
+  Priority priority = 3;
+  MessageType type = 4;
+  int64 epoch = 5;
+  UUID cluster_id = 6;
+  UUID client_id = 7;
+  bool ignore_cluster_id = 8;
+  bool ignore_epoch = 9;
+}
```

Comment:
Hi Zach, I'm wondering is there any specific reason we use underscore rather than camelCase for the fields of Header? I noticed that the generated Java code has the same format of getter methods though (like getRequestId() for request_id).

Reply:
Right now, there's a lack of consistency in our style that we need to address and update, but the recommendation is to use CamelCase for message names and using an underscore format for field names: https://confluence.eng.vmware.com/pages/viewpage.action?spaceKey=NSBU&title=NSBU+Protobuf+Conventions+and+Style+Guide and https://developers.google.com/protocol-buffers/docs/style

I haven't had a chance to make the necessary updates, but was thinking of making these changes when we incorporate Vjeko's feedback on the schema.

Reply:
Gotcha, thanks!